### PR TITLE
Order contains multiple qualities of a product

### PR DESCRIPTION
**Pull request by CarstenWalther**
This pull request updates the merchant to the new producer API changes. Mainly, an order from the producer contains a list of products in different qualities.

The example merchants store ordered products in a list. If a merchant creates a new offer, he takes products from that list. If the list is empty, the merchant orders new products to refill the list.

I also remove many producer API calls because merchants should not be allowed to reconfigure the producer.

### CODE REVIEW

File: pricewars_merchant.py
Comment (Bouncner) [inline pricewars_merchant.py]: We should try to get rid of `amount` when we actually talk about the `number of` or `counts`.

File: pricewars_merchant.py
Comment (Bouncner) [inline pricewars_merchant.py]: We have setting keys like `'restock limit'` and `'primeShipping'`. We should decide for either camel case or underscores (or even spaces, but we should unify all keys).

File: api/producer.py
Comment (Bouncner) [inline api/producer.py]: We shoud get rid of `amount` and rather use `count` or `number_of...`.

**Review by Bouncner — CHANGES_REQUESTED** [review]

**Comment by CarstenWalther** [comment]
I've also used underscore naming for strategy names to make it easier to use in terminal

**Review by Bouncner — APPROVED** [review]

**Comment by Bouncner** [comment]
Will merge in October, when the currently running semester has finished.

**Pull request merged**
